### PR TITLE
Introduced a AT transport group that supports atomic send and receive

### PR DESCRIPTION
**Pull request by SzymonPobiega**
* piggybacked a small refactoring in UnicastReplyRouterConnector

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@Particular/nservicebus-maintainers please review.

**Comment by bording** [comment]
Since you said the UnicastReplyRouterConnector was piggybacked, I'm assuming it's an unrelated change?

Personally, I'd rather not see unrelated changes grouped together like that. At the very least, I'd want it to be a separate commit, but I'd prefer to see it come in via separate PR.

Anyone disagree?

**Comment by andreasohlund** [comment]
>Since you said the UnicastReplyRouterConnector was piggybacked, I'm assuming it's an unrelated change?

Seems like a small refactoring to cleanup the code (please confirm @SzymonPobiega ) ,  but yes unrelated

>Personally, I'd rather not see unrelated changes grouped together like that. At the very least, I'd want it to be a separate commit, but I'd prefer to see it come in via separate PR.

>Anyone disagree?

Agree on separate pull if larger and at least a separate commit if small. How about this:

1 . @bording raise the above rule/guideline as a pull on - https://github.com/Particular/PlatformDevelopment/blob/master/Maintainers/nservicebus-maintainers.md?

2. We go ahead with this one due to the "smallness" of the change

3. Once agreed, we enforce the rule

Thoughts?

File: src/NServiceBus.Core/Routing/Routers/UnicastReplyRouterConnector.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ class UnicastReplyRouterConnector : StageConnector<IOutgoingReplyContext, IOutgo
 
             context.Headers[Headers.MessageIntent] = MessageIntentEnum.Reply.ToString();
 
-            var addressLabels = RouteToDestination(replyToAddress).EnsureNonEmpty(() => "No destination specified.").ToArray();
+            var addressLabels = new[] { new UnicastRoutingStrategy(replyToAddress)};
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Routing/Routers/UnicastReplyRouterConnector.cs]: Do we have a formatting issue here?

**Comment by danielmarbach** [comment]
@andreasohlund For me the key for a good PR

* It is mind-sized which means I can grasp the changes easily by reading the description and going over the changes in the PR
* It is focused on the topic of the PR description/title

**Comment by bording** [comment]
Now that #3308 is merged, it has changed how the ScenarioDescriptors work. This PR no longer compiles, so it will need to be updated to work.

It's now up to downstream `IConfigureTestExecution` implementations to define which descriptors should be excluded vs. including that logic in the descriptor itself.

File: src/NServiceBus.AcceptanceTests/ScenarioDescriptors/AllTransports.cs [hunk 2]
```diff
@@ -77,6 +77,19 @@ public AllTransportsWithoutNativeDeferral()
         }
     }
 
+    public class AllTransportsWithoutNativeDeferralAndAtomicSendAndReceive : AllTransports
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/ScenarioDescriptors/AllTransports.cs]: If I understand the logic of this filter correctly, I think a more descriptive name for this class would be

`AllTransportsWithoutNativeDeferralAndWithAtomicSendAndReceive`

**Comment by SzymonPobiega** [comment]
Rebased, removed that small refactoring and updated to the new behavior of descriptors.

**Comment by bording** [comment]
The only problem I see at this point is that the commit message still references the refactoring that has now been removed.

**Comment by SzymonPobiega** [comment]
Updated the commit message. Ready to be merged.

**Comment by danielmarbach** [comment]
@SzymonPobiega Are there any implications for the downstream transports? If yes can you associate the PRs here?

**Comment by SzymonPobiega** [comment]
@danielmarbach the only one directly affected is Rabbit but we have a local fix there so no need to immediately upgrade to latest core. We will switch to this fix in rabbit in the next update.

**Pull request merged**